mass_editing/models/mass_object.py

```diff
@@ -20,119 +20,116 @@
+from openerp import api, fields, models, _
 
 
-class MassObject(orm.Model):
+class MassObject(models.Model):
     _name = "mass.object"
 
-    _columns = {
-        'name': fields.char("Name", size=64, required=True, select=1),
-        'model_id': fields.many2one(
-            'ir.model', 'Model', required=True, select=1),
-        'field_ids': fields.many2many(
-            'ir.model.fields', 'mass_field_rel', 'mass_id', 'field_id',
-            'Fields'),
-        'ref_ir_act_window': fields.many2one(
-            'ir.actions.act_window', 'Sidebar Action', readonly=True,
-            help="Sidebar action to make this template available on records \
-                 of the related document model"),
-        'ref_ir_value': fields.many2one(
-            'ir.values', 'Sidebar Button', readonly=True,
-            help="Sidebar button to open the sidebar action"),
-        'model_ids': fields.many2many('ir.model', string='Model List')
-    }
+    name = fields.Char('Name', required=True, select=1)
+    model_id = fields.Many2one('ir.model', 'Model', required=True,
+                               help="Model is used for Selecting Fields."
+                               "This is editable until Sidebar menu is not "
+                               "created.")
+    field_ids = fields.Many2many('ir.model.fields', 'mass_field_rel',
+                                 'mass_id', 'field_id', 'Fields')
+    ref_ir_act_window = fields.Many2one('ir.actions.act_window',
+                                        'Sidebar action',
+                                        readonly=True,
+                                        help="Sidebar action to make this "
+                                        "template available on records of "
+                                        "the related document model.")
+    ref_ir_value = fields.Many2one('ir.values', 'Sidebar button',
+                                   readonly=True,
+                                   help="Sidebar button to open"
+                                   "the sidebar action")
+    model_list = fields.Char('Model List')
 
     _sql_constraints = [
         ('name_uniq', 'unique (name)', _('Name must be unique!')),
     ]
 
-    def onchange_model_id(self, cr, uid, ids, model_id, context=None):
-        if context is None:
-            context = {}
-        if not model_id:
-            return {'value': {'model_ids': [(6, 0, [])]}}
-        model_ids = [model_id]
-        model_obj = self.pool['ir.model']
-        active_model_obj = self.pool.get(model_obj.browse(
-            cr, uid, model_id).model)
-        if active_model_obj._inherits:
-            for key, val in active_model_obj._inherits.items():
-                found_model_ids = model_obj.search(
-                    cr, uid, [('model', '=', key)], context=context)
-                model_ids += found_model_ids
-        return {'value': {'model_ids': [(6, 0, model_ids)]}}
+    @api.onchange('model_id')
+    def onchange_model(self):
```

Comment:
The original `onchange_model_id` was a more appropriate name for this, as it directly matches the column. Although seems like it should be private too IMO: `_onchange_model_id`